paddle/fluid/framework/ir/graph.cc

```diff
@@ -61,6 +61,49 @@ Graph::Graph(const ProgramDesc &program) : program_(program) {
       var->inputs.push_back(node);
     }
   }
+
+  std::vector<ir::Node *> send_ops;
+  ir::Node *send_bar = nullptr;
+  std::vector<ir::Node *> recv_ops;
+  ir::Node *fetch_bar = nullptr;
+  for (ir::Node *node : Nodes()) {
```

Comment:
Should these in `graph.cc` or a single `pass`? or maybe we can move it later to those `Pass`

Reply:
Good Idea. We can do that in the future, including the split/concat ops.